core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -0,0 +1,202 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.Editable
+import android.text.Selection
+import android.text.Spannable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.webkit.WebView
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.view.ActionMode
+import org.kiwix.kiwixmobile.core.R
+
+class CompatFindActionModeCallback internal constructor(context: Context) :
+  ActionMode.Callback, TextWatcher, View.OnClickListener {
+  private val tag = "CompatFindActionMode"
+  @JvmField var isActive: Boolean = false
+
+  @SuppressLint("InflateParams")
+  private val customView: View = LayoutInflater.from(context).inflate(R.layout.webview_search, null)
+  private val editText: EditText = customView.findViewById(R.id.edit)
+  private val findResultsTextView: TextView = customView.findViewById(R.id.find_results)
+  private var webView: WebView? = null
+  private val input: InputMethodManager =
+    context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+  private var actionMode: ActionMode? = null
+
+  init {
+    editText.setOnClickListener(this)
+    setText("")
+  }
+
+  fun setActive() {
+    isActive = true
+  }
+
+  fun finish() {
+    actionMode?.finish()
+    webView?.clearMatches()
+  }
+
+  // Place text in the text field so it can be searched for.  Need to press
+  // the find next or find previous button to find all of the matches.
+  fun setText(text: String?) {
+    editText.setText(text)
+    val span: Spannable = editText.text
+    val length = span.length
+
+    // Ideally, we would like to set the selection to the whole field,
+    // but this brings up the Text selection CAB, which dismisses this
+    // one.
+    Selection.setSelection(span, length, length)
+
+    // Necessary each time we set the text, so that this will watch
+    // changes to it.
+    span.setSpan(this, 0, length, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+  }
+
+  // Set the WebView to search.  Must be non null, and set before calling startActionMode.
+  fun setWebView(webView: WebView?) {
+    this.webView = requireNotNull(webView) {
+      "WebView supplied to CompatFindActionModeCallback cannot be null"
+    }
+    findResultsTextView.visibility = View.VISIBLE
+    this.webView?.setFindListener { activeMatchOrdinal: Int, numberOfMatches: Int, _: Boolean ->
+      findResultsTextView.text = when {
+        editText.text.toString().isEmpty() -> {
+          ""
+        }
+        numberOfMatches == 0 -> {
+          "0/0"
+        }
+        else -> {
+          (activeMatchOrdinal + 1).toString() + "/" + numberOfMatches
```

Comment:
oh also we should fix this stupid java string concatenation and use kotlin string interpolation

Reply:
a02fb0f